server/game/gameSystems/DeckSearchSystem.ts

```diff
@@ -0,0 +1,284 @@
+import type { AbilityContext } from '../core/ability/AbilityContext';
+import { Card } from '../core/card/Card';
+import { EventName, Location, TargetMode } from '../core/Constants';
+import { DeckSearchEvent } from '../core/event/DeckSearchEvent';
+import { GameEvent } from '../core/event/GameEvent';
+import { GameSystem, IGameSystemProperties } from '../core/gameSystem/GameSystem';
+import { IPlayerTargetSystemProperties, PlayerTargetSystem } from '../core/gameSystem/PlayerTargetSystem';
+import Player from '../core/Player';
+import { shuffleArray } from '../core/utils/Helpers';
+
+type Derivable<T> = T | ((context: AbilityContext) => T);
+
+export interface IDeckSearchProperties extends IPlayerTargetSystemProperties {
+    targetMode?: TargetMode;
+    activePromptTitle?: string;
+    cardsToSearch?: number | ((context: AbilityContext) => number); // Number of cards to search through
+    selectCardCount?: number | ((context: AbilityContext) => number);
+    reveal?: boolean;
+    shuffle?: boolean | ((context: AbilityContext) => boolean);
+    title?: string;
+    immediateEffect?: GameSystem<IGameSystemProperties>; // What should be done with the selected cards?
+    message?: string;
+    uniqueNames?: boolean;
+    player?: Player;
+    choosingPlayer?: Player;
+    placeOnBottomInRandomOrder?: boolean;
```

Comment:
It looks like if it's true, it returns them to the bottom of the deck. If it's false, it's not clear to me what happens. I suspect they remain on top?

I was thinking we could eliminate this by making the remainingCardsHandler be set up to return the cards to the bottom by default. Thoughts?

Reply:
Yeah, looks like they remain on top. I think that is the best solution since afaik the only standard handling for the searched cards in SWU is randomized on bottom, otherwise it's a custom card action.